arch/risc-v/src/mpfs/mpfs_can.c

```diff
@@ -0,0 +1,2923 @@
+                              |CAN_SET_TSEG2(1)
+#define CAN_SPEED_16M_500K   CAN_SET_BITRATE(1)|CAN_SET_TSEG1(12) \
+                              |CAN_SET_TSEG2(1)
+#define CAN_SPEED_32M_500K   CAN_SET_BITRATE(3)|CAN_SET_TSEG1(12) \
+                              |CAN_SET_TSEG2(1)
+
+/* 25m         75% */
+#define CAN_SPEED_8M_1M      CAN_SET_BITRATE(0)|CAN_SET_TSEG1(4) \
+                              |CAN_SET_TSEG2(1)
+#define CAN_SPEED_16M_1M     CAN_SET_BITRATE(1)|CAN_SET_TSEG1(4) \
+                              |CAN_SET_TSEG2(1)
+#define CAN_SPEED_32M_1M     CAN_SET_BITRATE(3)|CAN_SET_TSEG1(4) \
+                              |CAN_SET_TSEG2(1)
+
+/* The following constants are used for error codes:
+ *
+ * |  Constants            |  Description                                |
+ * |-----------------------|---------------------------------------------|
+ * | CAN_OK                | Indicates there is no error                 |
+ * | CAN_ERR               | Indicates error condition                   |
+ * | CAN_TSEG1_TOO_SMALL   | Value provided to configure TSEG1 is too    |
+ * |                       | small                                       |
+ * | CAN_TSEG2_TOO_SMALL   | Value provided to configure TSEG2 is too    |
+ * |                       | small                                       |
+ * | CAN_SJW_TOO_BIG       | Value provided to configure synchronous jump|
+ * |                       | width (SJW) is too big.                     |
+ * | CAN_BASIC_CAN_BUFFER  | Indicates that buffer is configured for     |
+ * |                       | Basic CAN operation                         |
+ * | CAN_NO_RTR_BUFFER     | Indicates that there is no buffer for       |
+ * |                       | remote transmit request (RTR) frame         |
+ * | CAN_INVALID_BUFFER    | Indicates invalid buffer number             |
+ * | CAN_NO_MSG            | Indicates no message available              |
+ * | CAN_VALID_MSG         | Indicates message is valid                  |
+ */
+
+#define CAN_OK                  0
+#define CAN_ERR                 1
+#define CAN_TSEG1_TOO_SMALL     2
+#define CAN_TSEG2_TOO_SMALL     3
+#define CAN_SJW_TOO_BIG         4
+#define CAN_BASIC_CAN_BUFFER    5
+#define CAN_NO_RTR_BUFFER       6
+#define CAN_INVALID_BUFFER      7
+#define CAN_NO_MSG              8
+#define CAN_VALID_MSG           0
+
+/****************************************************************************
+ * Utility definitions
+ ****************************************************************************/
+
+#define print_uint32_t(prefix, val)  do { \
+  const char p_str[] = prefix; \
+  /* prefix + " 0b" + 32 bits + null terminator */ \
+  char binary_str[sizeof(p_str) + 2 + 32 + 1]; \
+  sprintf(binary_str, "%s 0b", p_str); \
+  for (int i = 31; i >= 0; i--) { \
+      sprintf(binary_str + strlen(binary_str), "%d", ((val) >> i) & 1); \
+  } \
+  caninfo("%s", binary_str); \
+} while (0)
+
+#if __GNUC__ >= 3
```

Comment:
@raiden00pl can correct me if I'm wrong, but I believe NuttX specifically avoids checking compiler/C/C++ versions outside of some of the core compiler code `compiler.h`?

Reply:
actually this part was a left over from a refactor previously and it's now not needed. I will remove